### PR TITLE
fix(app): fix choose csv file button label

### PR DESCRIPTION
**Pull request by koji**
# Overview
update the label for choose csv button, confirm values disabled condition and add `useFeatureFlag`
For `ChooseCSV` button, we need check two things. One is type and the other is analysis's reuslt.
if type is csv_file and result is ok => green background('ready') and display `file name`
if type is csv_file and result is parameter-value-required => background ('not ready') and display `required`

close AUTH-539

### before
Screenshot 2024-06-26 at 2 11 02 PM

### after
Screenshot 2024-06-26 at 4 44 01 PM

# Test Plan
set up a protocol run with the following protocol
[rtp_phase2_test.py.zip](https://github.com/user-attachments/files/15993624/rtp_phase2_test.py.zip)

# Changelog

# Review requests

# Risk assessment
low

### CODE REVIEW

File: app/src/organisms/ProtocolSetupParameters/index.tsx [hunk 1]
```diff
@@ -148,6 +153,14 @@ export function ProtocolSetupParameters({
     }
   }
 
+  console.log('isLoading', isLoading)
+  console.log('startSetup', startSetup)
+  console.log('enableCsvFile', enableCsvFile)
+  console.log(
+    'result',
+    mostRecentAnalysis?.result === 'parameter-value-required'
+  )
```
Comment (ncdiehl11) [inline app/src/organisms/ProtocolSetupParameters/index.tsx]: 🪵

File: app/src/organisms/ProtocolSetupParameters/index.tsx [hunk 2]
```diff
@@ -157,6 +170,10 @@ export function ProtocolSetupParameters({
         }}
         onClickButton={handleConfirmValues}
         buttonText={t('confirm_values')}
+        buttonIsDisabled={
+          enableCsvFile &&
+          mostRecentAnalysis?.result === 'parameter-value-required'
```
Comment (ncdiehl11) [inline app/src/organisms/ProtocolSetupParameters/index.tsx:167]: I refactored the `getAnalysisStatus` utility in [this PR](https://github.com/Opentrons/opentrons/pull/15527), but this works for now!

File: app/src/organisms/ProtocolSetupParameters/index.tsx (hunk 1 above)
Comment (koji) [reply app/src/organisms/ProtocolSetupParameters/index.tsx]: thanks, I'll remove them.

**Review by jerader — APPROVED** [review]
lgtm!

**Pull request merged**